edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -159,13 +204,31 @@ internal Task<Option<ServiceIdentity>> GetServiceIdentityFromService(string id)
             return cache;
         }
 
+        async Task<bool> ShouldRefreshIdentity(string id)
+        {
+            bool hasRefreshed = this.identitiesLastRefreshTime.TryGetValue(id, out DateTime lastRefreshTime);
```

Comment:
Do we also need to consider this.cacheLastRefreshTime?

Reply:
no, we shouldn't wait even in that case.  On-demand refresh should always be treated with a higher priority that cache refresh, because it's invoked when a new device connects, but we couldn't find the device's identity in the cache.  We can't predict how long into the delay timer we'll be if we start conflating the two scenarios, what if it's 1m59s since we last refreshed the cache?  What if the user changed the delay so that it's significantly longer?

IMO it's better to always keep individual vs cache refresh separate, both in terms of predictability and simplicity.